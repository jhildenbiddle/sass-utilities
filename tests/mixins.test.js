const compileSass = require('./helpers/compile-sass');

describe('fluid-prop', () => {
    test('Compile', () => {
        const result = compileSass(`
            .myclass {
                @include fluid-prop(font-size, 5vw, 16px, 48px);
            }
        `);

        expect(result).toMatchSnapshot();
    });
});

describe('mq', () => {
    test('Compile - Defaults', () => {
        const result = compileSass(`
            .myclass {
                @include mq(small) {
                    color: red;
                }
            }
        `);

        expect(result).toMatchSnapshot();
    });

    test('Compile - feature and value as list', () => {
        const result = compileSass(`
            .myclass {
                @include mq(max-width medium) {
                    color: green;
                }
            }
        `);

        expect(result).toMatchSnapshot();
    });

    test('Compile - multiple conditions via nessted queries', () => {
        const result = compileSass(`
            .myclass {
                @include mq(small) {
                    @include mq(landscape) {
                        color: red;
                    }
                    @include mq(portrait) {
                        color: blue;
                    }
                }
            }
        `);

        expect(result).toMatchSnapshot();
    });
});

describe('triangle', () => {
    test('Compile', () => {
        const result = compileSass(`
            .myclass {
                @include triangle();
            }
        `);

        expect(result).toMatchSnapshot();
    });
});
