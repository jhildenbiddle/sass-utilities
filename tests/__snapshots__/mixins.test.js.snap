// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fluid-prop Compile 1`] = `
".myclass {
  font-size: 5vw;
}
@media (max-width: 320px) {
  .myclass {
    font-size: 16px;
  }
}
@media (min-width: 960px) {
  .myclass {
    font-size: 48px;
  }
}"
`;

exports[`mq Compile - Defaults 1`] = `
"@media screen and (min-width: 30em) {
  .myclass {
    color: red;
  }
}"
`;

exports[`mq Compile - feature and value as list 1`] = `
"@media screen and (max-width: 48em) {
  .myclass {
    color: green;
  }
}"
`;

exports[`mq Compile - multiple conditions via nessted queries 1`] = `
"@media screen and (min-width: 30em) and (orientation: landscape) {
  .myclass {
    color: red;
  }
}
@media screen and (min-width: 30em) and (orientation: portrait) {
  .myclass {
    color: blue;
  }
}"
`;

exports[`triangle Compile 1`] = `
".myclass {
  content: "";
  width: 0;
  height: 0;
  border-top: 16px solid currentColor;
  border-bottom: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
}"
`;
